Module/randomize.py

```diff
@@ -54,6 +54,35 @@ def populateLocations(self, excludeWorlds):
 
         self._validLocationListDonald = [location for location in self._allLocationListDonald if not set(location.LocationTypes).intersection(excludeWorlds)]
 
+        late_item_weight = 1
+        early_item_weight = 1
+        insane = False
+        if item_difficulty == "Super Easy":
+            early_item_weight = 50
+            late_item_weight = .1
+        if item_difficulty == "Easy":
+            early_item_weight = 1
+            late_item_weight = .1
+        if item_difficulty == "Hard":
+            early_item_weight = 1
+            late_item_weight = 5
+        if item_difficulty == "Very Hard":
+            early_item_weight = .1
+            late_item_weight = 5
+        if item_difficulty == "Insane":
+            early_item_weight = .01
+            late_item_weight = 50
+            insane = True
+
+        for loc in self._validLocationList:
+            if loc.LocationWeight>1:
+                loc.setLocationWeight(late_item_weight)
+            elif loc.LocationWeight<1:
+                loc.setLocationWeight(early_item_weight)
+            if insane and locationType.DataOrg in loc.LocationTypes:
```

Comment:
Have you tested to see how the distribution looks for insane? This seems like you would end up with something absurd like 12+ checks on Data Org every time, which is maybe fine but maybe there should be a little more variation.

It would be nice to have a utility script that could make a seed with set settings, and then pretty print out the locations of just the important checks to make it easier to eyeball test the weights. Then we could decide how many data org locations should have ICs on average, and it would be straightforward to figure out the proper weight.

Reply:
Anecdotally, I wasn't getting any Data checks because there are only 8 data org slots (the second AS slots are not marked Data Org currently) and given that there are currently 111 "hard" locations marked, it was super unlikely to get  a check on them. By increasing those 8 locations to be 4x the other locations, it roughly made the chance of a data getting a check go from 8/111 to 32/135. I'll do what you suggested and build a way to estimate item distribution and we can tweak from there.